majifix311/src/main/java/com/example/majifix311/ui/ProblemDetailActivity.java

```diff
@@ -0,0 +1,167 @@
+package com.example.majifix311.ui;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.app.Activity;
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.widget.Toolbar;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.example.majifix311.R;
+import com.example.majifix311.models.Problem;
+import com.example.majifix311.ui.views.ExpandableImageView;
+import com.example.majifix311.utils.MapUtils;
+
+import java.util.Locale;
+
+public class ProblemDetailActivity extends Activity {
+    private static final String PROBLEM_INTENT = "problem";
+    private Problem mProblem;
+
+    private Toolbar mToolbar;
+    private ImageView mIvMap;
+    private ExpandableImageView mIvPhoto;
+    private TextView mTvReporter;
+    private TextView mTvCategory;
+    private TextView mTvLocation;
+    private TextView mTvDescription;
+
+    public static void startActivity(Activity callingActivity, Problem problem) {
+        Intent intent = new Intent(callingActivity, ProblemDetailActivity.class);
+        intent.putExtra(PROBLEM_INTENT, problem);
```

Comment:
This caused a `E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 1317440)`, which in turn caused `java.lang.RuntimeException: android.os.TransactionTooLargeException: data parcel size 1317440 bytes`. Don't push image data (likely to be more than the allowed 1MB) around in Intents, and therefore don't push Problems around in Intents. Since we already have a static pseudo-factory, maybe from here cache it as a static variable of this class, and then in onCreate() copy it to the member variable and null out the static reference?

See https://stackoverflow.com/questions/8552514/is-there-some-limits-in-android-bundle for reference.

Reply:
Darn! I was hoping that making the image smaller would avoid this problem, as I didn't see it after reducing the size, but I guess not. As this is the case, let me make a few changes to this PR so as to avoid this issue. The idea I have is to save the images, and then pass around the url with which they can be accessed.